provider/postgis/postgis.go

```diff
@@ -315,45 +316,38 @@ func BuildDBConfig(opts *DBConfigOptions) (*pgxpool.Config, error) {
 		return nil, err
 	}
 
-	dbconfig.ConnConfig.LogLevel = pgx.LogLevelWarn
 	dbconfig.ConnConfig.RuntimeParams = opts.GetRuntimeParams()
-	var hstore hstoreOID
+
+	// NOTE: reflects previous pgx/v4 behaviour of passing pgx.loglevelwarn
+	logAdapter := NewLoggerAdapter()
+	tracer := &tracelog.TraceLog{
+		Logger:   logAdapter,
+		LogLevel: tracelog.LogLevelWarn,
+	}
+	dbconfig.ConnConfig.Tracer = tracer
+
+	var hstoreOID uint32
 
 	dbconfig.AfterConnect = func(ctx context.Context, conn *pgx.Conn) error {
-		// The AfterConnect call runs everytime a new connection is acquired,
-		// including everytime the connection pool expands. The hstore OID
-		// is not constant, so we lookup the OID once per provider and store it.
-		// Extensions have to be registered for every new connection.
-
-		if !hstore.hasInit {
-			row := conn.QueryRow(ctx, "SELECT oid FROM pg_type WHERE typname = 'hstore';")
-			if err = row.Scan(&hstore.OID); err != nil {
-				switch {
-				case errors.Is(err, pgx.ErrNoRows):
-					// do nothing, because query can be empty if hstore is not installed
-					break
-				default:
-					return fmt.Errorf("error fetching hstore oid: %w", err)
-				}
+		row := conn.QueryRow(ctx, "SELECT oid FROM pg_type WHERE typname = 'hstore';")
+		if err = row.Scan(&hstoreOID); err != nil {
+			switch {
+			case errors.Is(err, pgx.ErrNoRows):
+				// do nothing, because query can be empty if hstore is not installed
+				break
+			default:
+				return fmt.Errorf("error fetching hstore oid: %w", err)
 			}
-			hstore.hasInit = true
 		}
 
-		// dont register hstore data type if hstore extension is not installed
-		if hstore.OID != 0 {
-			conn.ConnInfo().RegisterDataType(pgtype.DataType{
-				Value: &pgtype.Hstore{},
-				Name:  "hstore",
-				OID:   hstore.OID,
-			})
-		}
+		conn.TypeMap().RegisterType(&pgtype.Type{
+			Name:  "hstore",
+			OID:   hstoreOID,
+			Codec: pgtype.HstoreCodec{},
+		})
 
 		// register UUID type, see https://github.com/jackc/pgx/wiki/UUID-Support
-		conn.ConnInfo().RegisterDataType(pgtype.DataType{
-			Value: &gofrs.UUID{},
-			Name:  "uuid",
-			OID:   pgtype.UUIDOID,
-		})
+		pgxuuid.Register(conn.TypeMap())
```

Comment:
I was hoping to avoid by using the builtin pgtype codec for uuid, but I just couldn't get it to work without this external dependency.